main.tf

```diff
@@ -115,3 +119,13 @@ resource "aws_route53_record" "this" {
     evaluate_target_health = false
   }
 }
+
+resource "aws_cloudfront_function" "functions" {
+  count = length(var.cf_functions)
```

Comment:
Better to use `for_each` instead of count.

Reply:
The variable is a list and I think for_each does not work for the list type (at least for me), but I can change it like the following:
```
resource "aws_cloudfront_function" "functions" {
  for_each = {for func in var.cf_functions: func.type => func}

  name    = each.value.name
  runtime = "cloudfront-js-1.0"
  comment = each.value.comment
  publish = true
  code    = each.value.code
}
```

Reply:
yes, I meant that. but keys should be unique.
```
for_each = {for func in var.cf_functions: func.name => func}
```

Reply:
from the discussion it became clear that `func.type` should be an unique field.

Reply:
I have changed the data type of the input variable to map and have updated the relevant places